### DISCUSSION TITLE
Can I use pyinstaller to package projects only with code?

### DISCUSSION THREAD

**Discussion by lmcjrrg** [pyinstaller/pyinstaller#7526] [Help]
Hello, I have some trouble and need help

The machine environment version is as follows

pyinstaller:5.9.0
ubuntu:20.04
python:3.8.10
torch:1.12.1
cuda:11.6
deepstream:6.1

There is a dependent environment on the ubuntu system where the project can run normally. At this point, I only want to use pyinstaller to package the code in the project into an executable file, and the executable file only runs on the local machine. (If you package the dependent environment, such as torch, cuda, deepstream, etc., it will cause the packaged file to be super large, several gigabytes in size.)

**Can I achieve this if I only package the code? How should I configure it in. spec?**

My English is not very good, please forgive me

**Comment by rokm** [comment]
PyInstaller creates (or at least tries to create) self-contained executables with isolated environment, so no, you cannot use it to package just the code. 

#6022 implemented an option to remove system libraries from the PyInstaller-frozen application (see [here](https://pyinstaller.org/en/stable/spec-files.html#posix-specific-options)), but I doubt it will help you much, because all 3rd party python packages (such as `torch`) and the libraries that they bring with them will still be collected.

Why would you want to create executable in the first place? If you are running on local machine with full environment available, just run your python script directly...

**Reply by lmcjrrg** [reply]
Thank you for your reply,
Because the server is used externally, it is necessary to prevent others from seeing the source code I encrypted the file using cpython, and then packaged it using pyinstaller -- key

**Reply by rokm** [reply]
`pyinstaller --key` does very little to actually protect your code, because anyone can look at PyInstaller's source code and see how the encryption key is stored. And then create an automatic unpacker script. See #6999 (although `PyExtractor` mentioned there does not seem to be available anymore).

**Reply by lmcjrrg** [reply]
Okay, thank you for reminding me
